src/main/kotlin/ro/hibyte/betting/repository/UserProfileRepository.kt

```diff
@@ -8,4 +8,6 @@ import java.util.*
 interface UserProfileRepository : JpaRepository<UserProfile, Long>{
     fun findByKeycloakId(keycloakId: String): UserProfile?
     fun findByUsername(userGroupName: String): UserProfile
+    fun findByUsernameAndUserIdNot(username: String?, userId: Long?): Optional<UserProfile>
```

Comment:
return "UserProfile?", and will return null if not found

Optional is just a mind stretching exercise that is needed in Java, but not in Kotlin, since it has nullable types and build in Null-handling with the ? operator